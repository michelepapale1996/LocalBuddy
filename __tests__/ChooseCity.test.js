import ChooseCity from "../screens/ChooseCity"
import renderer from 'react-test-renderer';
import React from 'react';

let findById = function(tree, testID) {
    if(tree.props && tree.props.testID === testID) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], testID)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}


test('renders correctly at the beginning', () => {
    const tree = renderer.create(<ChooseCity/>).toJSON();
    expect(tree).toMatchSnapshot();

    const searchbarComponent = findById(tree, "searchbar")
    expect(searchbarComponent.props.placeholder).toBe("Search a city...")

    const chooseCityInstance = renderer.create(<ChooseCity/>).getInstance()
    expect(chooseCityInstance.state.initialCities.length).toBe(6)
});