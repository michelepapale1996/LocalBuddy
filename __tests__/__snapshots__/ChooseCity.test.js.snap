// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly at the beginning 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2fa1ff",
        "borderRadius": 4,
        "elevation": 4,
        "flexDirection": "row",
        "height": 187,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 18,
              "height": 36,
              "justifyContent": "center",
              "margin": 6,
              "overflow": "visible",
              "width": 36,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View
        style={null}
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <TextInput
      accessibilityRole="search"
      accessibilityTraits="search"
      allowFontScaling={true}
      keyboardAppearance="light"
      onChangeText={[Function]}
      placeholder="Search a city..."
      placeholderTextColor="white"
      rejectResponderTermination={true}
      returnKeyType="search"
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "flex": 1,
            "fontSize": 18,
            "minWidth": 0,
            "paddingLeft": 8,
            "textAlign": "left",
          },
          Object {
            "color": "#000000",
            "fontFamily": "Helvetica Neue",
          },
          Object {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
          },
        ]
      }
      testID="searchbar"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityStates={
        Array [
          "disabled",
        ]
      }
      accessible={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 18,
              "height": 36,
              "justifyContent": "center",
              "margin": 6,
              "overflow": "visible",
              "width": 36,
            },
            Object {
              "opacity": 0.32,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        style={null}
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(255, 255, 255, 0)",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "id": "ChIJdd4hrwug2EcRmSrV3Vo6llI",
            "name": "London",
            "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/london_web-e8502ca139.jpg",
          },
          Object {
            "id": "ChIJOwg_06VPwokRYv534QaPC8g",
            "name": "New York",
            "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/new-york_web-99d9bb0809.jpg",
          },
          Object {
            "id": "ChIJXSModoWLGGARILWiCfeu2M0",
            "name": "Tokyo",
            "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/tokyo_web-7a20b5733f.jpg",
          },
          Object {
            "id": "ChIJRcbZaklDXz4RYlEphFBu5r0",
            "name": "Dubai",
            "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/dubai_web-e211678fd9.jpg",
          },
          Object {
            "id": "ChIJIQBpAG2ahYAR_6128GcTUEo",
            "name": "San Francisco",
            "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/san-francisco-bay-area_web-f17b1f60e6.jpg",
          },
          Object {
            "id": "ChIJE9on3F3HwoAR9AhGJW_fL-I",
            "name": "Los Angeles",
            "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/los-angeles-f5b60deb04.jpg",
          },
        ]
      }
      disableVirtualization={false}
      extraData={
        Object {
          "cities": Array [],
          "initialCities": Array [
            Object {
              "id": "ChIJdd4hrwug2EcRmSrV3Vo6llI",
              "name": "London",
              "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/london_web-e8502ca139.jpg",
            },
            Object {
              "id": "ChIJOwg_06VPwokRYv534QaPC8g",
              "name": "New York",
              "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/new-york_web-99d9bb0809.jpg",
            },
            Object {
              "id": "ChIJXSModoWLGGARILWiCfeu2M0",
              "name": "Tokyo",
              "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/tokyo_web-7a20b5733f.jpg",
            },
            Object {
              "id": "ChIJRcbZaklDXz4RYlEphFBu5r0",
              "name": "Dubai",
              "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/dubai_web-e211678fd9.jpg",
            },
            Object {
              "id": "ChIJIQBpAG2ahYAR_6128GcTUEo",
              "name": "San Francisco",
              "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/san-francisco-bay-area_web-f17b1f60e6.jpg",
            },
            Object {
              "id": "ChIJE9on3F3HwoAR9AhGJW_fL-I",
              "name": "Los Angeles",
              "url": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/los-angeles-f5b60deb04.jpg",
            },
          ],
          "loadingDone": true,
          "noCitiesAvailable": false,
          "query": "",
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 4,
                "elevation": 10,
                "marginBottom": 13.5,
                "marginLeft": 7.5,
                "marginRight": 7.5,
                "marginTop": 13.5,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 9,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 10,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexGrow": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#eeeeee",
                      "height": 195,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    Object {
                      "height": 267,
                    },
                  ]
                }
              >
                <Image
                  siblings={
                    Array [
                      "withTheme(Card.Cover)",
                      "Card.Content",
                    ]
                  }
                  source={
                    Object {
                      "uri": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/london_web-e8502ca139.jpg",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "height": null,
                        "justifyContent": "flex-end",
                        "padding": 16,
                        "resizeMode": "cover",
                        "width": null,
                      },
                      Object {
                        "borderTopLeftRadius": 4,
                        "borderTopRightRadius": 4,
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "paddingBottom": 16,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Helvetica Neue",
                        "textAlign": "left",
                        "writingDirection": "ltr",
                      },
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "HelveticaNeue-Medium",
                          "textAlign": "left",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  London
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 4,
                "elevation": 10,
                "marginBottom": 13.5,
                "marginLeft": 7.5,
                "marginRight": 7.5,
                "marginTop": 13.5,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 9,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 10,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexGrow": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#eeeeee",
                      "height": 195,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    Object {
                      "height": 267,
                    },
                  ]
                }
              >
                <Image
                  siblings={
                    Array [
                      "withTheme(Card.Cover)",
                      "Card.Content",
                    ]
                  }
                  source={
                    Object {
                      "uri": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/new-york_web-99d9bb0809.jpg",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "height": null,
                        "justifyContent": "flex-end",
                        "padding": 16,
                        "resizeMode": "cover",
                        "width": null,
                      },
                      Object {
                        "borderTopLeftRadius": 4,
                        "borderTopRightRadius": 4,
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "paddingBottom": 16,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Helvetica Neue",
                        "textAlign": "left",
                        "writingDirection": "ltr",
                      },
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "HelveticaNeue-Medium",
                          "textAlign": "left",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  New York
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 4,
                "elevation": 10,
                "marginBottom": 13.5,
                "marginLeft": 7.5,
                "marginRight": 7.5,
                "marginTop": 13.5,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 9,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 10,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexGrow": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#eeeeee",
                      "height": 195,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    Object {
                      "height": 267,
                    },
                  ]
                }
              >
                <Image
                  siblings={
                    Array [
                      "withTheme(Card.Cover)",
                      "Card.Content",
                    ]
                  }
                  source={
                    Object {
                      "uri": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/tokyo_web-7a20b5733f.jpg",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "height": null,
                        "justifyContent": "flex-end",
                        "padding": 16,
                        "resizeMode": "cover",
                        "width": null,
                      },
                      Object {
                        "borderTopLeftRadius": 4,
                        "borderTopRightRadius": 4,
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "paddingBottom": 16,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Helvetica Neue",
                        "textAlign": "left",
                        "writingDirection": "ltr",
                      },
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "HelveticaNeue-Medium",
                          "textAlign": "left",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  Tokyo
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 4,
                "elevation": 10,
                "marginBottom": 13.5,
                "marginLeft": 7.5,
                "marginRight": 7.5,
                "marginTop": 13.5,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 9,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 10,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexGrow": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#eeeeee",
                      "height": 195,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    Object {
                      "height": 267,
                    },
                  ]
                }
              >
                <Image
                  siblings={
                    Array [
                      "withTheme(Card.Cover)",
                      "Card.Content",
                    ]
                  }
                  source={
                    Object {
                      "uri": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/dubai_web-e211678fd9.jpg",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "height": null,
                        "justifyContent": "flex-end",
                        "padding": 16,
                        "resizeMode": "cover",
                        "width": null,
                      },
                      Object {
                        "borderTopLeftRadius": 4,
                        "borderTopRightRadius": 4,
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "paddingBottom": 16,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Helvetica Neue",
                        "textAlign": "left",
                        "writingDirection": "ltr",
                      },
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "HelveticaNeue-Medium",
                          "textAlign": "left",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  Dubai
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 4,
                "elevation": 10,
                "marginBottom": 13.5,
                "marginLeft": 7.5,
                "marginRight": 7.5,
                "marginTop": 13.5,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 9,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 10,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexGrow": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#eeeeee",
                      "height": 195,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    Object {
                      "height": 267,
                    },
                  ]
                }
              >
                <Image
                  siblings={
                    Array [
                      "withTheme(Card.Cover)",
                      "Card.Content",
                    ]
                  }
                  source={
                    Object {
                      "uri": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/san-francisco-bay-area_web-f17b1f60e6.jpg",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "height": null,
                        "justifyContent": "flex-end",
                        "padding": 16,
                        "resizeMode": "cover",
                        "width": null,
                      },
                      Object {
                        "borderTopLeftRadius": 4,
                        "borderTopRightRadius": 4,
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "paddingBottom": 16,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Helvetica Neue",
                        "textAlign": "left",
                        "writingDirection": "ltr",
                      },
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "HelveticaNeue-Medium",
                          "textAlign": "left",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  San Francisco
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 4,
                "elevation": 10,
                "marginBottom": 13.5,
                "marginLeft": 7.5,
                "marginRight": 7.5,
                "marginTop": 13.5,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 9,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 10,
              }
            }
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexGrow": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#eeeeee",
                      "height": 195,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    Object {
                      "height": 267,
                    },
                  ]
                }
              >
                <Image
                  siblings={
                    Array [
                      "withTheme(Card.Cover)",
                      "Card.Content",
                    ]
                  }
                  source={
                    Object {
                      "uri": "https://d13k13wj6adfdf.cloudfront.net/urban_areas/los-angeles-f5b60deb04.jpg",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "height": null,
                        "justifyContent": "flex-end",
                        "padding": 16,
                        "resizeMode": "cover",
                        "width": null,
                      },
                      Object {
                        "borderTopLeftRadius": 4,
                        "borderTopRightRadius": 4,
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "paddingHorizontal": 16,
                    },
                    Object {
                      "paddingBottom": 16,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Helvetica Neue",
                        "textAlign": "left",
                        "writingDirection": "ltr",
                      },
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontFamily": "HelveticaNeue-Medium",
                          "textAlign": "left",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                        Array [
                          Object {
                            "fontSize": 20,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  Los Angeles
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#52c8ff",
        "borderRadius": 28,
        "bottom": 0,
        "elevation": 6,
        "margin": 16,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 6,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 28,
          },
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "height": 56,
              "width": 56,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
